subsys/net/lib/aws_iot/src/aws_iot.c

```diff
@@ -1206,27 +1244,16 @@ void aws_iot_cloud_poll(void)
 	aws_iot_evt.type = AWS_IOT_EVT_CONNECTING;
 	aws_iot_notify_event(&aws_iot_evt);
 
-	err = client_broker_init(&client);
-	if (err) {
-		LOG_ERR("client_broker_init, error: %d", err);
-	}
-
-	err = mqtt_connect(&client);
-	if (err) {
-		LOG_ERR("mqtt_connect, error: %d", err);
-	}
-
-	err = connect_error_translate(err);
-
+	err = connect_client(NULL);
```

Comment:
I think the changes to how the connection is established should be in a separate commit